src/indexer/index_writer.rs

```diff
@@ -515,7 +515,7 @@ impl IndexWriter {
     /// Merges a given list of segments
     ///
     /// `segment_ids` is required to be non-empty.
-    pub fn merge(&mut self, segment_ids: &[SegmentId]) -> FutureResult<SegmentMeta> {
```

Comment:
Can we return `FutureResult<Option<SegmentMeta>>` ?

It can be very useful for people building tooling.